### ISSUE TITLE
Question about the controllers

### ISSUE THREAD

**Issue by ChristianSchneeweiss** [dev-mastery/comments-api#8]
Why do you inject the `add-comment` in the controller `post-comment`. Should inner layers not be imported into outer layers usually? I can not wrap my head around this. On all other layers you imported the inner ones and injected the outer layers.

**Comment by arcdev1** [comment]
You're right. Good catch. `add-comment` can be imported and does not need to be injected.

**Comment by restjohn** [comment]
Strictly speaking, Uncle Bob's Clean Architecture diagram shows that the Use Case Interactor implements a Use Case Input Port interface, and the Controller references the Input Port interface.  While importing the Use Case Interactor directly in the Controller does not violate the dependency rule, injecting an Input Port implementation does make testing the controller easier.
